docs/topics/coroutines.rst

```diff
@@ -42,10 +42,71 @@ hence use coroutine syntax (e.g. ``await``, ``async for``, ``async with``):
     methods of
     :ref:`downloader middlewares <topics-downloader-middleware-custom>`.
 
+-   The
+    :meth:`~scrapy.spidermiddlewares.SpiderMiddleware.process_start_requests`
+    method of :ref:`spider middlewares <custom-spider-middleware>`.
+
 -   :ref:`Signal handlers that support deferreds <signal-deferred>`.
 
+-   The :meth:`~scrapy.spiders.Spider.start_requests` spider method.
+
 .. _asynchronous generators were introduced in Python 3.6: https://www.python.org/dev/peps/pep-0525/
 
+.. _async-start_requests:
+
+Asynchronous start_requests and spider middlewares
+==================================================
+
+.. versionadded:: 2.1
+
+The :meth:`~scrapy.spiders.Spider.start_requests` spider method can be an
+asynchronous generator. In this case all spider middlewares used with this
+spider that have the
+:meth:`~scrapy.spidermiddlewares.SpiderMiddleware.process_start_requests`
+method must support this: if they receive an asynchronous iterable, they must
+return one as well. On the other hand, if they receive a normal iterable, they
+shouldn't break and ideally should return a normal iterable too. There can be
```

Comment:
Why is it recommended to keep non-async iterables as they are? Is it because of a risk that some other middleware doesn't support async yet, or are there other reasons (e.g. performance)?